More details surrounding `JEditTextArea`

Stefterv opened the pull request.

Stefterv committed:
Update README.md

SableRaf commented on app/src/processing/app/syntax/README.md:
"~~the~~ Jetpack Compose" -> "Jetpack Compose"

SableRaf commented on app/src/processing/app/syntax/README.md:
I want to add questions about feature usage to the next community survey and I can include Tweak mode and autocomplete. Let me know if there are others you'd like me to add.

SableRaf commented on app/src/processing/app/syntax/README.md:
```diff
@@ -1,4 +1,14 @@
-# 🐉 Fixing this code: here be dragons. 🐉
+# Replacing our custom version of JEditTextArea
+
+As we have started the migration to Jetpack Compose we will eventually need to replace the JEditTextArea as well.
```
This lacks a bit of context as a first paragraph.

SableRaf commented on app/src/processing/app/syntax/README.md:
```diff
@@ -1,4 +1,14 @@
+
+As we have started the migration to Jetpack Compose we will eventually need to replace the JEditTextArea as well.
+
+I think a good current strategy would be to start using `RSyntaxTextArea` for the upcoming p5.js mode as it is a better maintained well rounded library. As noted below, a lot of the current state management of the PDE is interetwined with the JEditTextArea implementation. This will force us to decouple the state management out of the `JEditTextArea` whilst also trying to keep backwards compatibil ...
```
Good idea!

To make the sentence clearer, consider specifying that "RSyntaxTextArea is a better maintained, well-rounded library," since it's not immediately obvious whether that description refers to `RSyntaxTextArea` or p5.js.

Also, it might help to clarify that you're proposing to use the upcoming Electron-based p5.js mode as a test bed for integrating `RSyntaxTextArea` into the PDE. This is somewhat implied but it could be clearer.

Stefterv committed:
Revise README for Jetpack Compose migration strategy

Updated README to reflect migration to Jetpack Compose and strategy for replacing JEditTextArea with RSyntaxTextArea. Added insights on LSP-based editor research and the need for user feedback on Tweak Mode and autocompletion features.

SableRaf reviewed (approved):
lgtm
